ext2/sb.c

```diff
@@ -58,27 +56,23 @@ int ext2_read_sb(id_t *devId, ext2_fs_info_t *f)
 
 int ext2_write_sb(ext2_fs_info_t *f)
 {
-	int ret = 0, err;
-
-	ret = atasrv_write(&f->devId, EXT2_SB_OFF, (char *)f->sb, EXT2_SB_SZ, &err);
+	ssize_t ret;
 
-	if (ret != EXT2_SB_SZ) {
-		printf("ext2: superblock write error %d\n", ret);
+	if ((ret = atasrv_write(&f->devId, EXT2_SB_OFF, (char *)f->sb, EXT2_SB_SZ)) != EXT2_SB_SZ)
 		return -EFAULT;
-	}
 
 	return EOK;
 }
 
 
 void ext2_init_fs(id_t *devId, ext2_fs_info_t *f)
 {
-	int size, err;
+	int size;
 	void *buff;
 
 	f->block_size = EXT2_SB_SZ << f->sb->log_block_size;
 	f->blocks_count = f->sb->blocks_count;
-	printf("ext2: Mounting %.2f MiB partition\n", (float)(f->block_size * f->blocks_count)/1024/1024);
+	printf("ext2: Mounting %u B partition\n", f->block_size * f->blocks_count);
```

Comment:
It will overflow on partitions bigger than 4 GiB.

Reply:
Fixed by using PRIu64 for printing uint64_t values.